cpp/include/raft/neighbors/ball_cover-inl.cuh

```diff
@@ -255,9 +254,9 @@ void all_knn_query(raft::resources const& handle,
  *               looking in the closest landmark.
  * @param[in] n_query_pts number of query points
  */
-template <typename idx_t, typename value_t, typename int_t>
+template <typename idx_t, typename value_t, typename int_t, typename matrix_idx = std::uint32_t>
```

Comment:
I was looking through the impact of adding the `matrix_idx` everywhere and I was about to ask that we not add it to these functions before I realized it's already on the `BallCoverIndex` and you are just exposing it everywhere. Still, I think in a future change we should consider removing it. Just making a note here about it.